### ISSUE TITLE
Authorization required

### ISSUE THREAD

**Issue by ashwinvis** [YunoHost-Apps/syncthing_ynh#28]
I did a fresh install of yunohost on raspberry pi, did a `apt upgrade` and installed syncthing. On launching the app a pop up screen launches and says authorization required. What could be wrong?

**Comment by yalh76** [comment]
Can you provide some printscreens ?

**Comment by ashwinvis** [comment]
![Screenshot_2019-08-11-09-01-26](https://user-images.githubusercontent.com/9155111/62835290-0c891d80-bc57-11e9-8fe4-854b962bd6b5.png)

The screenshot (from mobile browser). I was trying to access Syncthing as a user from the SSO page. I saw #3 that you can only access it as the admin. Is that the reason?

**Comment by yalh76** [comment]
Yes automatic authentication is not implemented, you have to provide another time your YunoHost username and password

**Comment by yalh76** [comment]
It works ? I can close the issue ?

**Comment by yalh76** [comment]
I can't test it I don't have an android phone...

**Comment by ashwinvis** [comment]
I just gave the screenshot via the phone browser. It is hosted on a Raspberry Pi actually.

**Comment by pabaillon** [comment]
Hello, same thing for me, fresh install on raspberry, installation of syncthing with the yunohost web interface. It ask me  "Authorization Required" with login and password...

**Comment by yalh76** [comment]
> 
> 
> Hello, same thing for me, fresh install on raspberry, installation of syncthing with the yunohost web interface. It ask me "Authorization Required" with login and password...

And what happens when you provide a yunohost user and password ?

**Comment by pabaillon** [comment]
The login windows come back. 
"Le site  demande un nom d’utilisateur et un mot de passe. Le site indique : « Authorization Required »"
After retry, a blank page with
"Not Authorized"

**Comment by zbam** [comment]
Same issue here with the same message "Le site https://chezdouteau.hopto.org demande un nom d’utilisateur et un mot de passe. Le site indique : « Authorization Required »"

**Comment by zbam** [comment]
![Screen Shot 2019-08-14 at 21 00 50](https://user-images.githubusercontent.com/1504536/63048261-ac4be300-bec5-11e9-9991-80a00bf14bf0.png)
Same issue here with the same message "Le site https://*****.hopto.org demande un nom d’utilisateur et un mot de passe. Le site indique : « Authorization Required »"

**Comment by zbam** [comment]
![Screen Shot 2019-08-14 at 21 00 50](https://user-images.githubusercontent.com/1504536/63048384-f7fe8c80-bec5-11e9-966d-20b60502e8be.png)
Same issue here with the same message "Le site https://*****.hopto.org demande un nom d’utilisateur et un mot de passe. Le site indique : « Authorization Required »"

**Comment by yalh76** [comment]
OK, I've tested. Installing a YunoHost from scratch, adding a domain, installing a certificate, creating a yunohost user. 

Then when I go to syncthing, it asks for a login and password. I provide the login and password previously created, and I'm allowed to connect.

Can you provide the logs after a failed authentication doing `journalctl -u syncthing | yunopaste`

**Comment by pabaillon** [comment]
I did
`sudo journalctl -u syncthing | sudo /usr/bin/yunopaste`
I get
[https://paste.yunohost.org/hamiminaho](https://paste.yunohost.org/hamiminaho)

**Comment by yalh76** [comment]
OK then, seems a ldap issue. Can you provide the content of `/var/www/syncthing/.config/syncthing/config.xml` doing `more /var/www/syncthing/.config/syncthing/config.xml | sudo /usr/bin/yunopaste`

And provide `service slapd status | sudo /usr/bin/yunopaste` for the sldap service status

**Comment by pabaillon** [comment]
config 
https://paste.yunohost.org/uhojabirej

LDAP statut
https://paste.yunohost.org/higatayabe

**Comment by yalh76** [comment]
well, LDAP seems OK.
Can you try to replace `<address>localhost:389</address>` by `<address>127.0.0.1:389</address>` in `/var/www/syncthing/.config/syncthing/config.xml` and restart the syncthing service.

**Comment by pabaillon** [comment]
Thanks for your reply but it doestn't work

**Comment by yalh76** [comment]
Can you try to install phpldapadmin_ynh and try to access your yunohost ldap to check if it works

**Comment by pabaillon** [comment]
Done,  ldap seems to works with phpldapadmin_ynh

**Comment by yalh76** [comment]
dammm.

On what kind of material have you install syncthing ? Raspberry PI ? VPS ? to you have more informations to provide ?

**Comment by pabaillon** [comment]
Hi,
I try again with a Fresh install of yunohost-stretch-3.4.2-rpi-stable on raspberry pi 3 model B
creating a yunohost user.
Update of yunohost (if not I can't install syncthing)
Install of syncthing
Same issue

**Comment by andresolnik** [comment]
is there any workaround for this bug?

**Comment by andresolnik** [comment]
deleting `config.xml` file did the trick.

**Comment by Zuhlf** [comment]
> deleting `config.xml` file did the trick.

Deleting "config.xml" did NOT do the trick. It opens the configuration page of Syncthing to everyone, without any authentification (anybody can access to yourdomain.tld/syncthing and configure any syncronization with your files...)

**Comment by milomindrbindr** [comment]
> > deleting `config.xml` file did the trick.
> 
> Deleting "config.xml" did NOT do the trick. It opens the configuration page of Syncthing to everyone, without any authentification (anybody can access to yourdomain.tld/syncthing and configure any syncronization with your files...)
> 
> @yalh76 : could you have a look please?

Didn't work for me. Having the same issue on the current build.

**Comment by Gia90** [comment]
I am also experiencing the same issue.
After some research, it seems it is due to a bug in the `gopkg.in/ldap.v2` lib used in Syncthing ([link](https://github.com/go-ldap/ldap/pull/199)).
The fix would require to upgrade that library to the `gopkg.in/ldap.v3`.

Here more info:
https://github.com/portainer/portainer/issues/3244
https://github.com/nonylene/go-check-plugins/commit/7eaa7ac05a0527cf513fabd7618f70caf40f0d5c

**Comment by chunjiw** [comment]
Same issue here.

**Comment by tituspijean** [comment]
Temporary workaround, if you are only one user needing Syncthing, taking into account the issue raised above (https://github.com/YunoHost-Apps/syncthing_ynh/issues/28#issuecomment-569654689):

1. In the webadmin, make your user the only one with rights to access the app.
2. In `/var/www/syncthing/.config/syncthing/config.xml`, delete the `ldap` line within the `<gui>...</gui>` tags, and remove the whole `<ldap>...</ldap>` that is right after it.
3. `systemctl restart syncthing.service`

This way, Syncthing does not try to contact the LDAP server, and the SSO takes care of blocking out unwanted users.

**Comment by propellaone** [comment]
> This way, Syncthing does not try to contact the LDAP server, and the SSO takes care of blocking out unwanted users.

Doesn't work for me. I can access syncthing now, but so can everybody else. SSO doesn't block unauthorized users.

**Comment by tituspijean** [comment]
You are right, sorry about that. I was on YNH v3.7 testing branch.

**Comment by Pfeil** [comment]
I am also affected. Running YunoHost 3.7.0.12 (stable) on a Raspberry Pi 3 with a fresh install of the current image on the website, then fully updated and after that I began installing actual applications. Everything is running smooth except this issue, so I do not assume that something went wrong or this setup is unusual in some kind of way. Do I see that correctly that not everyone is able to reproduce the issue?

**Comment by tituspijean** [comment]
Syncthing's author has updated the dependency, so let's cross our fingers that its next release fixes our issue :)

**Comment by tituspijean** [comment]
Unfortunately, the issue is still present with latest release:

```
Apr 23 21:15:15 rpi.zt syncthing[441]: 2020/04/23 21:15:15 ldap: recovered panic in processMessages: runtime error: invalid memory address or nil pointer dereference
Apr 23 21:15:15 rpi.zt syncthing[441]: [QPVQY] WARNING: LDAP Start TLS: LDAP Result Code 200 "Network Error": ldap: response channel closed
```

My config is:
```
    <ldap>
        <address>localhost:389</address>
        <bindDN>uid=%s,ou=users,dc=yunohost,dc=org</bindDN>
    </ldap>
```

I also tried replacing `localhost` with `127.0.0.1` and `<transport>starttls</transport>`, or `<insecureSkipVerify>true</insecureSkipVerify>`, to no avail.

**Comment by Pfeil** [comment]
By latest version you refer to 1.5.0-rc.2, right? Well, that is disappointing. But nice to hear about the permission integration. I think there was an option for more verbose logging somewhere, but I did not find it... maybe we should try to get more logging information and report an issue at syncthing with those, if it does not help. It might still be a problem on their side.

**Comment by tituspijean** [comment]
No, I tested v1.4.2, which was released after the "fix" was committed.

**Comment by yalh76** [comment]
I've just tested on a fresh YunoHost install and there is no ldap issue with default parameters/ When I logon on the web interface, it asks me to authenticate and after providing login and password I'm on the interface

**Comment by Pfeil** [comment]
@tituspijean Uh now I am confused (I never used the release feature of github). Looking at the commit name of 1.4.2 and the 1.4 branch, the fix might not be in the 1.4 release, but according to the master branch it would be...

@yalh76 did you try on a raspberry pi 3 model b?

anyway: I reinstalled syncthing on my ynh and was not able to login. May I ask which username/pw exactly is needed? So I chose a user at installation named firstname lastname (user@domain). Is the login name "user" or "firstname lastname" or "user@domain" or "admin" ...? Just to be sure and to simplify tests.

**Comment by milomindrbindr** [comment]
> 
> 
> I've just tested on a fresh YunoHost install and there is no ldap issue with default parameters/ When I logon on the web interface, it asks me to authenticate and after providing YunoHost login and password I'm on the web interface

Weird, just updated everything to the latest versions and still not able to login with any admin or user login and password.

**Comment by yalh76** [comment]
> @yalh76 did you try on a raspberry pi 3 model b?

No I tried on X64_64, I don't have a raspberry pi 3

> anyway: I reinstalled syncthing on my ynh and was not able to login. May I ask which username/pw exactly is needed? So I chose a user at installation named firstname lastname (user@domain). Is the login name "user" or "firstname lastname" or "user@domain" or "admin" ...? Just to be sure and to simplify tests.

Just tested, any YunoHost User using:
user
password

Works for me...

**Comment by yalh76** [comment]
> 
> 
> > I've just tested on a fresh YunoHost install and there is no ldap issue with default parameters/ When I logon on the web interface, it asks me to authenticate and after providing YunoHost login and password I'm on the web interface
> 
> Weird, just updated everything to the latest versions and still not able to login with any admin or user login and password.

On which plateform do you installed it ?

**Comment by tituspijean** [comment]
So, I have at my disposal three types of devices:
* OVH SSD VPS: had the issue, deleted the app and reinstalled, no issue any more.
* Docker container: never got the issue. Can log in with the prompt sent by Syncthing itself, or through YNH.
* Raspberry Pi 3B: issue remains after deleting the app and reinstalling it (making sure no configuration file remains after deletion)

**Comment by Pfeil** [comment]
Things changed for me. I think it is fixed.
When logged in into yunohost, I can access without typing any password. It scared me at first, but I can not access when not logged in, so it seems to be fine. I haven't actively used it yet, but it seems to work properly.

The only thing I did was installing the recent updates from the yunohost menu a few days ago. There were some libldap updates I think, and a yunohost update, as well as a syncthing update. I installed everything before testing (I had no hope :D) so I do not know which update fixed it.
